bundles/com.espressif.idf.ui/src/com/espressif/idf/ui/InitializeToolsStartup.java

```diff
@@ -102,25 +106,32 @@ public void run()
 		}
 		else if (isInstallerConfigSet())
 		{
-			checkForUpdatedVersion(idf_json_file);
-			if (isInstallerConfigSet())
+			if (!isVersionDifferentInFile(idf_json_file))
 			{
-				Logger.log("Ignoring esp_idf.json settings as it was configured earilier and idf_path is similar.");
 				return;
 			}
 
+			Logger.log("A different version for idf tool in ide env and esp_idf.json found");
 			IDFEnvironmentVariables idfEnvMgr = new IDFEnvironmentVariables();
 			Display.getDefault().syncExec(() -> {
 				Shell shell = new Shell(Display.getDefault());
 				MessageBox messageBox = new MessageBox(shell, SWT.ICON_WARNING | SWT.YES | SWT.NO);
 				messageBox.setText(Messages.ToolsInitializationDifferentPathMessageBoxTitle);
 				messageBox.setMessage(MessageFormat.format(Messages.ToolsInitializationDifferentPathMessageBoxMessage,
-						newIdfPath, idfEnvMgr.getEnvValue(IDFEnvironmentVariables.IDF_PATH)));
+						idfEnvMgr.getEnvValue(IDFEnvironmentVariables.IDF_PATH)));
 				int response = messageBox.open();
-				if (response == SWT.NO)
+				if (response == SWT.YES)
 				{
-					IDFEnvironmentVariables idfEnvironmentVariables = new IDFEnvironmentVariables();
-					updateEspIdfJsonFile(idf_json_file, idfEnvironmentVariables.getEnvValue(IDFEnvironmentVariables.IDF_PATH));
+					String version = IDFUtil.getEspIdfVersion();
+					java.util.regex.Pattern p = java.util.regex.Pattern.compile("([0-9][.][0-9])"); //$NON-NLS-1$
+					java.util.regex.Matcher m = p.matcher(version);
+					if (m.find())
+					{
+						version = m.group(0);
+					}
+					updateEspIdfJsonFile(idf_json_file, idfEnvMgr.getEnvValue(IDFEnvironmentVariables.IDF_PATH),
+							IDFUtil.getIDFPythonEnvPath(),
+							version);
```

Comment:
What if the user updated esp-idf(let's say 5.0 to 5.1) through windows installer, in this case `isVersionDifferentInFile` is `true` considering we compare esp-idf path from IDE and esp-idf.json

Don't we export the paths in this case?